cmd/docker/docker.go

```diff
@@ -279,12 +282,45 @@ func isSupported(cmd *cobra.Command, details versionDetails) error {
 			if _, ok := f.Annotations["experimentalCLI"]; ok && !hasExperimentalCLI {
 				errs = append(errs, fmt.Sprintf("\"--%s\" is only supported when experimental cli features are enabled", f.Name))
 			}
+			if _, ok := f.Annotations["kubernetes"]; ok && !hasKubernetes {
+				errs = append(errs, fmt.Sprintf("\"--%s\" is only supported on a Docker cli with kubernetes features enabled", f.Name))
+			}
+			if _, ok := f.Annotations["swarm"]; ok && hasKubernetes {
+				errs = append(errs, fmt.Sprintf("\"--%s\" is only supported on a Docker cli with swarm features enabled", f.Name))
```

Comment:
```
$ docker service ls
docker service ls is only supported on a Docker cli with swarm features enabled
```

Perhaps we should change that error a bit; because using `docker info` will still show that Swarm is enabled;

```
Swarm: active
 NodeID: zqa3qal0mc4smzwp0jgmgkzy4
 Is Manager: true
```

"with the swarm orchestrator" enabled? Or (for the experimental stage/beta) 

    "The Docker CLI is currently configured to use kubernetes for orchestration. Configure 'swarm' as orchestrator to use the swarm commands" 

(suggestions welcome)